File: src/app/models/LoanPayment.js

// Define collection and schema for LoanPayment
export interface LoanPayment {
    paymentReference:
	type : string
    amount:
	type : Money
    paymentDate:
	type : Date
    LoanAccount:
	type : Schema.Types.ObjectId
    Transaction:
	type : Schema.Types.ObjectId
    Method:
 	type : String
    Status:
 	type : String
#
    collection: 'loanPayments'
}

File: bankingOnGolang/internal/model/LoanPayment.go
package model

import (
	"time"
)

// ==============================================================
// LoanPayment Declaration
// ==============================================================
type LoanPayment struct {
	BaseModel
	PaymentReference string
	Amount           Money `gorm:"embedded;embeddedPrefix:loanPayment_amount"`
	PaymentDate      time.Time
	LoanAccountId    *uint
	LoanAccount      *LoanAccount `gorm:"foreignKey:LoanAccountId"`
	TransactionId    *uint
	Transaction      *Transaction `gorm:"foreignKey:TransactionId"`
	Method           PaymentMethod
	Status           PaymentStatus

	// parent associations as their child
	PaymentsFromLoanAccountId *uint
}
